src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,7 +106,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            if(!Files.isWritable(path)) {
```

Comment:
No need to check for the writability explicitly; you can make use the standard exception thrown from Java